src/common/base/abstractReportOnPromote.ts

```diff
@@ -49,33 +49,27 @@ export abstract class ReportOnPromoteCommand<T extends typeof SfCommand> extends
       : (this.flags['job-id'] as string);
     const org = this.flags['devops-center-username'] as Org;
 
-    try {
-      // query the deployment result related to the promote operation.
-      const operationResult: DeploymentResult = await selectOneDeploymentResultByAsyncJobId(
-        org.getConnection(),
-        asyncJobId
+    // query the deployment result related to the promote operation.
+    const operationResult: DeploymentResult | null = await selectOneDeploymentResultByAsyncJobId(
+      org.getConnection(),
+      asyncJobId
+    );
+    if (operationResult && !this.jsonEnabled()) {
+      const displayer: OutputService = new OutputServiceFactory().forPromotionReport(
+        this.flags,
+        operationResult,
```

Comment:
```suggestion
        operationResult!,
```

Then I think you need this to correct the type of operationResult

Reply:
Typescript should understand that operationResult is not undefined, given the guard testing operationResult. Using `!` is an anti-pattern that should be avoided.

Reply:
In this case operationResult it exists or it's null at this point, because we previously tested the existence inside the query method and we send back the correct result.

This is the return from the query method:
`  return resp.totalSize > 0 ? resp.records[0] : null;`